### DISCUSSION TITLE
How to restore Form state when javascript is disabled?

### DISCUSSION THREAD

**Discussion by cdeutsch** [remix-run/remix#2591] [Q&A] [answered]
I was testing out the ability to disable javascript and have Forms still work.

The messages returned in `actionData` are visible, but the Form `<input>` values used are lost.

I could return the current Form values as part of `actionData`, but I was wondering if there's another way to do it?

**Comment by sergiodxa** [comment, marked as answer]
The other way would be to store them in the session and return them from the loader.

I think returning them from the action is easier
